backend/src/packages/auth/auth.service.ts

```diff
@@ -75,8 +75,11 @@ class AuthService {
     };
   }
 
-  public getAuthenticatedUser(id: number): Promise<UserAuthResponseDto | null> {
-    return this.userService.findById(id);
+  public async getAuthenticatedUser(id: number): Promise<UserAuthResponseDto> {
+    const user = (await this.userService.findById(id)) as UserAuthResponseDto;
```

Comment:
```suggestion
    const user = (await this.userService.findById(id)) as UserAuthResponseDto;
```
What if this.userService.findById(id) returns null? Will it work as expected?

Reply:
@HIMDanny Good explanation!

Reply:
this relates on business logic too much, it's better not to expect only user here, but null also

Reply:
ok, then I'll change it